plugins/modules/nmcli.py

```diff
@@ -1650,7 +1650,7 @@ def connection_options(self, detect_change=False):
                 'vxlan.local': self.vxlan_local,
                 'vxlan.remote': self.vxlan_remote,
             })
-        elif self.type == 'wifi':
+        elif self.type in ('wifi', '802-11-wireless'):
```

Comment:
This change should not be needed, since it only translates between the Ansible _.yml_ file and the actual `nmcli` command, which is handled correctly already.

Reply:
After putting `insert pdb; pdb.set_trace()` in a useful location:

```
826198775# python3 AnsiballZ_nmcli.py execute
> /root/.ansible/tmp/ansible-tmp-1677030194.5205107-802761-74731826198775/debug_dir/ansible_collections/community/general/plugins/modules/nmcli.py(1588)connection_options()
-> if self.type == 'bond':
(Pdb) p self.type
'802-11-wireless'
(Pdb)
```

This seems to contradict your claim that the change is not necessary.

Reply:
As far as I got it, `self.type` here takes your _ansible.yml_ configuration setting, which should be `wifi` according to the docs, and translates it into `802-11-wireless` or `802-11-wireless-security` for the `nmcli` command.

Using `epdb`...

```
        elif self.type == 'wifi':
            options.update({
                '802-11-wireless.ssid': self.ssid,
                'connection.slave-type': 'bond' if self.master else None,
            })
            if self.wifi:
                for name, value in self.wifi.items():
                    options.update({
                        '802-11-wireless.%s' % name: value
                    })
            if self.wifi_sec:
                for name, value in self.wifi_sec.items():
                    options.update({
                        '802-11-wireless-security.%s' % name: value
                    })
            import epdb; epdb.serve()
````

Then connecting
```
# epdb
Waiting for a socket to appear at localhost:8080
> /tmp/ansible_nmcli_payload_necfii0n/ansible_nmcli_payload.zip/ansible/modules/nmcli.py(1710)connection_options()
(Epdb) print(self.type)
wifi

(Epdb) print(options)
{'connection.autoconnect': True, 'connection.zone': None, 'ipv4.addresses': None, 'ipv4.dhcp-client-id': None, 'ipv4.dns': None, 'ipv4.dns-search': None, 'ipv4.ignore-auto-dns': False, 'ipv4.gateway': None, 'ipv4.ignore-auto-routes': False, 'ipv4.routes': None, 'ipv4.route-metric': None, 'ipv4.routing-rules': None, 'ipv4.never-default': False, 'ipv4.method': None, 'ipv4.may-fail': True, 'ipv6.addresses': None, 'ipv6.dns': None, 'ipv6.dns-search': None, 'ipv6.ignore-auto-dns': False, 'ipv6.gateway': None, 'ipv6.ignore-auto-routes': False, 'ipv6.routes': None, 'ipv6.route-metric': None, 'ipv6.method': None, 'ipv6.ip6-privacy': None, 'ipv6.addr-gen-mode': None, 'connection.master': None, '802-11-wireless.ssid': 'Hotspot', 'connection.slave-type': None, '802-11-wireless.wake-on-wlan': 6}
```

My Ansible _test.yml_.
```
---
- name: "Test nmcli"
  hosts: ren

  tasks:
  - name: "Set Wake on Wifi"
    nmcli:
      type: wifi
      wifi:
        wake-on-wlan: 6
      conn_name: "Hotspot"
      ssid: "Hotspot"
      state: present
```

Maybe you want to share your test details and/or a user with more experience on this module should review this.

Reply:
Here is the key piece of code in nmcli.py:

```
        if not self.type:
            current_con_type = self.show_connection().get('connection.type')
            if current_con_type:
                self.type = current_con_type
```

If the user does not specify the connection type explicitly (and my use case does not), it is pulled from the output of nmcli, which returns `802-11-wireless`, not `wifi`. Perhaps the better fix would be to convert `802-11-wireless` to `wifi` here rather than checking for `802-11-wireless` elsewhere?

Reply:
Good find. =0)
I agree that it would be better to convert `802-11-wireless` to `wifi` here when `self.type` is not set. 
Do you just want to include this in this PR?